### ISSUE TITLE
Difficulty using `manage.py sqlflush`

### ISSUE THREAD

**Issue by stefanfoulis** [django-salesforce/django-salesforce#244]
When running `manage.py sqlflush` an error occurs. `sqlflush` is used to empty all tables.

```
django.core.management.base.CommandError: Database project_name couldn't be flushed. Possible reasons:
  * The database isn't running or isn't configured correctly.
  * At least one of the expected database tables doesn't exist.
  * The SQL was invalid.
Hint: Look at the output of 'django-admin sqlflush'. That's the SQL this command wasn't able to run.
The full error: relation "PricebookEntry" does not exist
```

Adding `managed=False` to the `Meta` on all salesforce models seems to have made the error go away for me. I don't know what other implications that has though. 

We saw this issue appear when we upgraded from `django-salesforce` `0.7.2` to `0.8.1`.
Experienced with `Django==1.11.23`.

`managed=False` was removed between `v0.7.2` and `v0.8.1`: [compare](https://github.com/django-salesforce/django-salesforce/compare/v0.7.2...v0.8.1#diff-42313000fef79ece345ab4faa3ae38ebL82)

**Comment by hynekcer** [comment]
I can't reproduce this error exactly. The command `manage.py sqlflush` works usually and the default non-salesforce database like with a `--database=default` parameter. That is used for the ten usual Django tables for sessions, content types, auth, permissions etc.

The output of command 
`python manage.py sqlflush --database=salesforce` 
is empty with Django 1.11 and the command 
`python manage.py flush --database=salesforce` 
does nothing.

If someone uses a setting `SALESFORCE_DB_ALIAS = 'default'` to run fast simple tests locally then the command sqlflush works correctly also.

The command sqlflush doesn't make sense for Salesforce.
Salesforce is an SOQL database not compatible enough with SQL and there doesn't exist an SQL command for delete, insert, update. The only way to delete something in Salesforce must be done by two request. One REST API request to get primary keys by a request very similar to SQL query and second very different REST request to delete objects with these primary keys. The command `sqlflush` is impossible therefore.

The command `flush` would be very dangerous on Salesforce: 
1) It is not possible to easily restore the database from a backup because different primary keys would be created and all foreign keys must be updated to these.
2) Many tables can't be flushed e.g. users, profiles and permissions, otherwise the access to Salesforce would be lost. There is no way to specify individual tables by flush command.

Tables can be flushed individually e.g. this way currently:

    Contact.objects.filter(pk__gt='').delete()
---

It seems that a simple delete should be also supported now.

    Contact.objects.all().delete()

It was not supported initially because of caution. The command sqlflush could report a clear message that it is unsupported with Salesforce.

**Comment by stefanfoulis** [comment]
oops, sorry. The failing command was actually `manage.py flush` not `manage.py sqlflush`.

Thanks for the detailed reply.

The issue appeared with our setup when no `DATABASES["salesforce"]` is defined and also no `SALESFORCE_DB_ALIAS`. So perhaps we are using an invalid configuration starting with `0.8.0`.

**Comment by hynekcer** [comment]
That configuration is valid and I still can not reproduce any problem. It is probably related more to something in your project. A minimalistic example could be necessary to demonstrate the problem on it.

I fixed the command `flush` for Django >= 2.0 and also I fixed the command `SomeModel.objects.all().delete()`. It is inspired by my previous comment and not related directly to your problem.
